src/vcs/git.js

```diff
@@ -70,6 +70,9 @@ export async function checkoutRevision(config, cwd, repository, ref, rev) {
 
   // Fixes issue with local master checkout
   await run(`${config.git} merge ${remote}/${ref}`, { cwd });
+
+  // Force a repack to avoid changes when tarring up later
+  await run(`${config.git} repack -d -l`, { cwd });
```

Comment:
nice! probably not an issue running this every time a client wants to checkout a revision since it happens so quickly.

In the future I wonder if we want to add some things here for minimizing the size of a repo proper to packing it up:
https://github.com/taskcluster/taskcluster-vcs/blob/2d499fdc33f89c2963b1ead200edecbd292e1fe9/src/artifacts.js#L227